### PR TITLE
#250 test get single work package endpoint

### PR DESCRIPTION
**Pull request by JasperTheDog**
## Changes

Added 3 test for checking that exceptions are properly handled by the function getSingleWorkPackage, using jest spyOn

## Notes

I am not sure how to craft the result for the working test

## Test Cases
Done: a project wbs number is given
Done: the work package does not exist
Done: the project does not exist
Need Help: it works as intended 

## To Do

- [x] Create data for example return result when function works properly
- [x] Finish test for function working properly

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

Ticket #250

### CODE REVIEW

File: src/backend/tests/work-packages.test.ts [hunk 1]
```diff
@@ -157,4 +159,34 @@ describe('Work Packages', () => {
...
+  test('getSingleWorkPackage fails if a project wbs number given (work_package number of 0)', async () => {
+    await expect(
+      async () => await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 0 })
+    ).rejects.toThrow(new HttpException(404, 'WBS Number 1.1.0 is a project WBS#, not a Work Package WBS#'));
+  });
+
+  // test('getSingleWorkPackage runs properly', async () => {
+  //   jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(prismaWorkPackage1);
+
+  //   const result = await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 1 });
```
Comment (ryanhowe28) [inline src/backend/tests/work-packages.test.ts:188]: Wouldn't passing in 1.1.1 throw an error like your first test? You have to pass in the values that would return you prismaWorkPackage1

File: src/backend/src/transformers/work-packages.transformer.ts [hunk 2]
```diff
@@ -36,9 +36,8 @@ const workPackageTransformer = (wpInput: Prisma.Work_PackageGetPayload<typeof wo
       detail: change.detail,
       dateImplemented: change.dateImplemented
     })),
-    projectName: wpInput.project.wbsElement.name,
-    stage: wpInput.stage || undefined
-  } as WorkPackage;
```
Comment (anthonybernardi) [inline src/backend/src/transformers/work-packages.transformer.ts:41]: i think you deleted this by accident while fixing merge conflicts

File: src/backend/tests/work-packages.test.ts [hunk 3]
```diff
@@ -157,4 +160,34 @@ describe('Work Packages', () => {
...
+  test('getSingleWorkPackage fails if the work package does not exist', async () => {
+    jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(null);
+
+    await expect(
+      async () => await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 1 })
+    ).rejects.toThrow(new NotFoundException('Work Package', '1.1.1'));
+  });
+
+  test('getSingleWorkPackage fails if the project does not exist', async () => {
+    jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(null);
```
Comment (anthonybernardi) [inline src/backend/tests/work-packages.test.ts:173]: you don't need this test, it's not testing anything new

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice work! small things then good to go

**Pull request closed without merging**